src/main/scala/mesosphere/marathon/core/task/bus/MarathonTaskStatus.scala

```diff
@@ -31,12 +32,20 @@ object MarathonTaskStatus {
   case class Starting(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
   case class Running(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
 
+  // Note that we need to distinguish between TemporarilyUnreachable tasks and really lost tasks that won't come back
+  // tl;dr: don't use Lost
+  case class Lost(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
+
   sealed trait Terminal extends MarathonTaskStatus {
     override def terminal: Boolean = true
     def killed: Boolean = false
   }
   object Terminal {
     def unapply(terminal: Terminal): Option[Terminal] = Some(terminal)
+    def unapply(status: MarathonTaskStatus): Option[MarathonTaskStatus] = status.mesosStatus match {
```

Comment:
Not really. There is one usage in the `AppTaskLauncherActor`:
```
  case TaskStatusUpdate(_, taskId, MarathonTaskStatus.Terminal(_)) =>
```
where the match case expects a `MarathonTaskStatus`. I used the MesosTaskStatus underneath to make sure the unapply only returns a `MarathonTaskStatus.Terminal` if it's really considered gone ... I can't think of a cleaner way to achieve this here without refactoring legacy code.